Sources/DVR/Filter.swift

```diff
@@ -0,0 +1,50 @@
+//
+//  Filter.swift
+//  DVR
+//
+//  Created by Jáir Myree on 6/18/21.
+//  Copyright © 2021 Venmo. All rights reserved.
+//
+
+import Foundation
+
+public struct Filter {
+    var headers : [String : String]
+    var queryParameters : [String : String]
+    var postDataParameters : [String : String]
+    var beforeRecordResponse : ((inout Foundation.URLResponse, inout Data?) -> ())
+    var beforeRecordRequest : ((inout URLRequest) -> ())
+    
+    public init(headers: [String] = [], queryParameters: [String] = [], postDataParameters: [String] = [], replacer: String = "Redacted", requestHook : @escaping ((inout URLRequest) -> ()) = {_ in } , responseHook : @escaping ((inout Foundation.URLResponse, inout Data?) -> ()) = {_,_  in } ) {
+        var adjustedHeaders : [String : String] = [:]
+        var adjustedQueryParameters : [String : String] = [:]
+        var adjustedPostDataParameters : [String : String] = [:]
+        
+        for header in headers {
+            adjustedHeaders[header] =  replacer
+        }
+        
+        for queryParam in queryParameters {
+            adjustedQueryParameters[queryParam] = replacer
+        }
+        
+        for postDataParam in postDataParameters {
+            adjustedPostDataParameters[postDataParam] = replacer
+        }
```

Comment:
I'd recommend we add some unit tests for these if we are supporting the functionality.

Reply:
We would need to add unit tests anyhow to get this merged in the main repo, and you should get some practice writing unit tests :)

Reply:
I have the tests within my dummy project so I'll move them over.